Client side error level for master

wbadam opened the pull request:
Vaadin 8 implementation of #9816. Cherry picked changes and added compatibility package component changes and tests.

Resolves #3139

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/9817)
<!-- Reviewable:end -->

wbadam committed:
Create UI test

wbadam committed:
Add client side error level

wbadam committed:
Create utility class for error messages

wbadam committed:
Add error level to component state

wbadam committed:
Add error level converter

wbadam committed:
Set error level in state

wbadam committed:
Add component error level style

wbadam committed:
Add error level for tooltip

wbadam committed:
Add error level style name to button error indicator

wbadam committed:
Add error level style name to native button error indicator

wbadam committed:
Add error level style name to check box error indicator

wbadam committed:
Add error level style name to link error indicator

wbadam committed:
Add error level style name to slot error indicator

wbadam committed:
Set error level for ordered layout's slot

wbadam committed:
Add error level style name to form layout error indicator

wbadam committed:
Add error level style name to caption error indicator

wbadam committed:
Add error level style name to tab sheet error indicator

wbadam committed:
Add error level style name to accordion error indicator

wbadam committed:
Add error level style name to panel error indicator

wbadam committed:
Text field error level theme

wbadam committed:
Error indicator theme

wbadam committed:
Add date field UI test

wbadam committed:
Combo box error level theme

wbadam committed:
TwinColSelect error level theme

wbadam committed:
Tooltip error level theme

wbadam committed:
Error level on compatibility components

wbadam committed:
Create automated UI test for error level

tsuoanttila commented:
There's a compilation error in the uitest package.

wbadam committed:
Fix compilation error in automatic ui test

wbadam commented:
Fixed before but forgot to commit/push, now I did

---

Review status: 0 of 35 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817#-:-Kr_IqZwZHELQjHw_LUZ:b5u50lr)*
<!-- Sent from Reviewable.io -->

wbadam commented:
Review status: 0 of 35 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kr_WChytA6b0_Ot93Fq:-Kr_WChytA6b0_Ot93Fr:bzfgbhk) (no related file):*
Build failed but reference screenshot `LabelTooltipTest-testLabelTooltip_LINUX_PhantomJS_1_tooltipVisible.png` seems wrong to me rather than what the test produces.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

ahie commented:
Review status: 0 of 35 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[themes/src/main/themes/VAADIN/themes/valo/shared/_variables.scss, line 215 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks2xMVzGWZDLyRutwtJ:-Ks2xMVzGWZDLyRutwtK:b-ca2k65) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/themes/src/main/themes/VAADIN/themes/valo/shared/_variables.scss#L215)):*
> ```SCSS
>  * @type color
>  */
> $v-error-indicator-level-info-color: #00a7f5 !default;
> ```

These constants don't seem to appear elsewhere in the project so I am unsure where these are taken from. Ping @jouni for an opinion.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

ahie commented:
Reviewed 4 of 35 files at r1.
Review status: 4 of 35 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3Jq_ftOS9PV2WzM3A:-Ks3Jq_ftOS9PV2WzM3B:b-qz6khx) (no related file):*
A cleaner and less error prone solution would be to have some shared class that encapsulates both the error message and error level, although taking that approach would result in many backwards incompatible changes.

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 86 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3CjK0ANnjy6cVqMnA:-Ks3CjK0ANnjy6cVqMnB:bbjby4s) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/TooltipInfo.java#L86)):*
> identifier

missing javadoc

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 113 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3CvFIVR8qMA5u15Vs:-Ks3CvFIVR8qMA5u15Vt:bfvxfln) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/TooltipInfo.java#L113)):*
> public TooltipInfo

Missing javadoc

---

*[client/src/main/java/com/vaadin/client/VCaption.java, line 268 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3D7eozqAOHW6nga8r:-Ks3D7eozqAOHW6nga8s:b-1gyedp) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/VCaption.java#L268)):*
> ErrorUtil.STYLE_NAME_ERROR_INDICATOR

Could have this constant in `StyleConstants`. Not blocking.

---

*[client/src/main/java/com/vaadin/client/VErrorMessage.java, line 62 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3ERQOvZmMD5v_DVYB:-Ks3ERQOvZmMD5v_DVYC:b-ba948c) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/VErrorMessage.java#L62)):*
> ```Java
>     }
> 
>     public void updateErrorLevel(ErrorLevel errorLevel) {
> ```

Missing javadoc (also include empty since tag).

---

*[client/src/main/java/com/vaadin/client/ui/VAccordion.java, line 370 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3Q0PWDdq4TBij5GaF:-Ks3Q0PWDdq4TBij5GaG:bv0f2tl) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/ui/VAccordion.java#L370)):*
> ""

Should this be `tabState.iconAltText`?

---

*[client/src/main/java/com/vaadin/client/ui/button/ButtonConnector.java, line 54 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3QE2fz7R04xbhc-WZ:-Ks3QE2fz7R04xbhc-W_:b-mh9vls) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/ui/button/ButtonConnector.java#L54)):*
> "errorLevel"

Could probably add a separate method for this, seeing that the method is named `setErrorMessage`.

---

*[shared/src/main/java/com/vaadin/shared/ui/ErrorLevel.java, line 26 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3NqQPud0MzNdGpl5Q:-Ks3NqQPud0MzNdGpl5R:bj7mh8) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/shared/src/main/java/com/vaadin/shared/ui/ErrorLevel.java#L26)):*
> Error code

I would change these to "Error level" to stay consistent.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

ahie commented:
Reviewed 1 of 35 files at r1.
Review status: 5 of 35 files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3eUoB4hS-ix-B2ReX:-Ks3eUoCPdHmiBieJbjW:bp441iq) (no related file):*
There are several client-side components that have a member field `errorIndicatorElement`. A `HasErrorIndicator` interface has been added in 8.0. We could possibly consider adding the `ErrorUtil.setErrorLevelStyle` functionality there and refactor those calls higher up in the component connector hierarchy.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

wbadam committed:
Javadoc, minor changes

wbadam commented:
Review status: 5 of 35 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3Jq_ftOS9PV2WzM3A:-KtBTMB8F-2SEMdUrNne:b-vda6aq) (no related file):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

A cleaner and less error prone solution would be to have some shared class that encapsulates both the error message and error level, although taking that approach would result in many backwards incompatible changes.
</blockquote></details>

At first I tried a similar way but wanted to be completely backwards compatible

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 86 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3CjK0ANnjy6cVqMnA:-KtBNeGZ1CdWfeQ_3qPc:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/TooltipInfo.java#L86)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

> identifier

missing javadoc
</blockquote></details>

Done.

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 113 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3CvFIVR8qMA5u15Vs:-KtBOhiUrdqs8bM5MX8_:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/TooltipInfo.java#L113)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

> public TooltipInfo

Missing javadoc
</blockquote></details>

Done.

---

*[client/src/main/java/com/vaadin/client/VCaption.java, line 268 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3D7eozqAOHW6nga8r:-KtBPmLJzxcXFPxWPcSf:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/VCaption.java#L268)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

> ErrorUtil.STYLE_NAME_ERROR_INDICATOR

Could have this constant in `StyleConstants`. Not blocking.
</blockquote></details>

Done.

---

*[client/src/main/java/com/vaadin/client/VErrorMessage.java, line 62 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3ERQOvZmMD5v_DVYB:-KtBQHQflu2XtNfSx256:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/VErrorMessage.java#L62)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

Missing javadoc (also include empty since tag).
</blockquote></details>

Done.

---

*[client/src/main/java/com/vaadin/client/ui/VAccordion.java, line 370 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3Q0PWDdq4TBij5GaF:-KtBQgg5YjQ76sWTBDAi:baf3fp9) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/ui/VAccordion.java#L370)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

> ""

Should this be `tabState.iconAltText`?
</blockquote></details>

Sure, why not? :)

---

*[client/src/main/java/com/vaadin/client/ui/button/ButtonConnector.java, line 54 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3QE2fz7R04xbhc-WZ:-KtBRMLCbTZiZbYORnvo:bezj42y) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/ui/button/ButtonConnector.java#L54)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

> "errorLevel"

Could probably add a separate method for this, seeing that the method is named `setErrorMessage`.
</blockquote></details>

Renamed the method to `setErrorMessageAndLevel()`, do you think it's all right like that? Just want to minimise duplicated code

---

*[shared/src/main/java/com/vaadin/shared/ui/ErrorLevel.java, line 26 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3NqQPud0MzNdGpl5Q:-KtBRqWJv_Q629UHlvFC:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/shared/src/main/java/com/vaadin/shared/ui/ErrorLevel.java#L26)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

> Error code

I would change these to "Error level" to stay consistent.
</blockquote></details>

Done.

---

*[themes/src/main/themes/VAADIN/themes/valo/shared/_variables.scss, line 215 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks2xMVzGWZDLyRutwtJ:-KtBS8AGlRnSyGeFkuC7:b-pugl0k) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/themes/src/main/themes/VAADIN/themes/valo/shared/_variables.scss#L215)):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

These constants don't seem to appear elsewhere in the project so I am unsure where these are taken from. Ping @jouni for an opinion.
</blockquote></details>

This was designed by @tuomasnironen specifically for the different error levels.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

wbadam committed:
Move ErrorLevel to shared

wbadam committed:
Refactor error indicator handling

wbadam commented:
Review status: 1 of 52 files reviewed at latest revision, 10 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3eUoB4hS-ix-B2ReX:-KuFHGcWVALfmaxrpHpA:btbbywp) (no related file):*
<details><summary><i>Previously, ahie (Aleksi Hietanen) wrote…</i></summary><blockquote>

There are several client-side components that have a member field `errorIndicatorElement`. A `HasErrorIndicator` interface has been added in 8.0. We could possibly consider adding the `ErrorUtil.setErrorLevelStyle` functionality there and refactor those calls higher up in the component connector hierarchy.
</blockquote></details>

Refactored the error indicator element handling. Introduced `HasErrorIndicatorElement` interface to be implemented by widgets that have error indicator and moved the logic that was possible to a more centralised place. The logic that was not unifiable also got clearer I think.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Theme review not done yet.

---

Reviewed 6 of 35 files at r1, 8 of 16 files at r3, 31 of 34 files at r4.
Review status: 42 of 52 files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuTMa51GZcArPMnDcPt:-KuTMa52jIZmU59Ms-WG:boobhzz) (no related file):*
Are the changes in v7 compatibility package the same changes as in the 7.7 branch of this change? I'd use still the old ErrorLevel where possible in the v7 version, just to make it easier to port changes from 7 to compatibility package.

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 201 at r3](https://reviewable.io:443/reviews/vaadin/framework/9817#-KtGYVdPbSnmGZIRAPvE:-KtGYVdPbSnmGZIRAPvF:b5fubzl) ([raw file](https://github.com/vaadin/framework/blob/90ae98bc7d74e53398f034393433844d14971e6e/client/src/main/java/com/vaadin/client/TooltipInfo.java#L201)):*
> ```Java
>     }
> 
>     public ErrorLevel getErrorLevel() {
> ```

javadoc

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 205 at r3](https://reviewable.io:443/reviews/vaadin/framework/9817#-KtGYX09LsJe6k9SnP2J:-KtGYX09LsJe6k9SnP2K:b5fubzl) ([raw file](https://github.com/vaadin/framework/blob/90ae98bc7d74e53398f034393433844d14971e6e/client/src/main/java/com/vaadin/client/TooltipInfo.java#L205)):*
> ```Java
>     }
> 
>     public void setErrorLevel(ErrorLevel errorLevel) {
> ```

javadoc

---

*[client/src/main/java/com/vaadin/client/ui/AbstractComponentConnector.java, line 660 at r4](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuTJCGoLDvGTeAIet6O:-KuTJCGoLDvGTeAIet6P:b-o3915m) ([raw file](https://github.com/vaadin/framework/blob/8c4e766671f3d7241d17d164b56c9cc8a94219ab/client/src/main/java/com/vaadin/client/ui/AbstractComponentConnector.java#L660)):*
> ```Java
> 
>         // add or remove error level style name
>         ErrorUtil.setErrorLevelStyle(getWidget().getElement(),
> ```

Could this part be done in the `setErrorLevel()` where the widgets with `HasErrorIndicatorElement` are handled as well?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817#-:-KuTMr7TuDmzfmfdIqOr:b-66enkl)*
<!-- Sent from Reviewable.io -->

wbadam committed:
Add javadoc for tooltip info methods

wbadam committed:
Move widget style handling to its own method

wbadam commented:
Review status: 40 of 52 files reviewed at latest revision, 14 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuTMa51GZcArPMnDcPt:-KuUpKngKZLR_3KwCgja:bdkcyw3) (no related file):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

Are the changes in v7 compatibility package the same changes as in the 7.7 branch of this change? I'd use still the old ErrorLevel where possible in the v7 version, just to make it easier to port changes from 7 to compatibility package.
</blockquote></details>

Compatibility package is the same as the v7 version.

Do you mean that the `ErrorMessage.ErrorLevel` enum should remain and marked as `@deprecated`? The import is only changed in `Buffered.SourceException`, `Validator.InvalidValueException`, `Grid` and `AbstractLegacyComponentDeclarativeTest`.

Would it be possible to change
`import com.vaadin.server.ErrorMessage.ErrorLevel` to
`import com.vaadin.shared.ui.ErrorLevel;`
with the migration tool instead?

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 201 at r3](https://reviewable.io:443/reviews/vaadin/framework/9817#-KtGYVdPbSnmGZIRAPvE:-KuUrfHHHXp4D7ePcmcN:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/90ae98bc7d74e53398f034393433844d14971e6e/client/src/main/java/com/vaadin/client/TooltipInfo.java#L201)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

javadoc
</blockquote></details>

Done.

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 205 at r3](https://reviewable.io:443/reviews/vaadin/framework/9817#-KtGYX09LsJe6k9SnP2J:-KuUrfeRdo2-3jcTFoya:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/90ae98bc7d74e53398f034393433844d14971e6e/client/src/main/java/com/vaadin/client/TooltipInfo.java#L205)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

javadoc
</blockquote></details>

Done.

---

*[client/src/main/java/com/vaadin/client/ui/AbstractComponentConnector.java, line 660 at r4](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuTJCGoLDvGTeAIet6O:-KuUu8UA1xxTgq3h8bLX:b52614) ([raw file](https://github.com/vaadin/framework/blob/8c4e766671f3d7241d17d164b56c9cc8a94219ab/client/src/main/java/com/vaadin/client/ui/AbstractComponentConnector.java#L660)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

Could this part be done in the `setErrorLevel()` where the widgets with `HasErrorIndicatorElement` are handled as well?
</blockquote></details>

Sure thing

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 86 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3CjK0ANnjy6cVqMnA:-KuZOIleOyFUQYtKJ710:bg7igkv) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/TooltipInfo.java#L86)):*
<details><summary><i>Previously, wbadam (Adam Wagner) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Review status: 40 of 52 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuTMa51GZcArPMnDcPt:-KuZOjMFjgPEpL9HinUK:bvo35hy) (no related file):*
<details><summary><i>Previously, wbadam (Adam Wagner) wrote…</i></summary><blockquote>

Compatibility package is the same as the v7 version.

Do you mean that the `ErrorMessage.ErrorLevel` enum should remain and marked as `@deprecated`? The import is only changed in `Buffered.SourceException`, `Validator.InvalidValueException`, `Grid` and `AbstractLegacyComponentDeclarativeTest`.

Would it be possible to change
`import com.vaadin.server.ErrorMessage.ErrorLevel` to
`import com.vaadin.shared.ui.ErrorLevel;`
with the migration tool instead?
</blockquote></details>

Remain, marked as deprecated, moved into v7 compatibility package if possible.

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 113 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3CvFIVR8qMA5u15Vs:-KuZOOx1jT91I8HkRSyz:b32wtd) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/TooltipInfo.java#L113)):*
<details><summary><i>Previously, wbadam (Adam Wagner) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

ok

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 261 at r5](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuYT1QXiWPk2V9S0ZNo:-KuYT1QYYCwXilEcrA0Z:b-a0f1dq) ([raw file](https://github.com/vaadin/framework/blob/030d01b34670012e2a7b25529e2f842ce124185f/client/src/main/java/com/vaadin/client/TooltipInfo.java#L261)):*
> ```Java
>      * Gets the error level.
>      *
>      * @return the error level
> ```

still missing since tags for the new methods

---

*[client/src/main/java/com/vaadin/client/VErrorMessage.java, line 62 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3ERQOvZmMD5v_DVYB:-KuZOTAIM6A31CJosFQA:b32wtd) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/VErrorMessage.java#L62)):*
<details><summary><i>Previously, wbadam (Adam Wagner) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

ok

---

*[client/src/main/java/com/vaadin/client/ui/VAccordion.java, line 370 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3Q0PWDdq4TBij5GaF:-KuZOabnWTRSCAlRkvuI:b32wtd) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/ui/VAccordion.java#L370)):*
<details><summary><i>Previously, wbadam (Adam Wagner) wrote…</i></summary><blockquote>

Sure, why not? :)
</blockquote></details>

ok

---

*[client/src/main/java/com/vaadin/client/ui/button/ButtonConnector.java, line 54 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3QE2fz7R04xbhc-WZ:-KuZOd82Xwxk9Ui0iEib:b-10pig7) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/client/src/main/java/com/vaadin/client/ui/button/ButtonConnector.java#L54)):*
<details><summary><i>Previously, wbadam (Adam Wagner) wrote…</i></summary><blockquote>

Renamed the method to `setErrorMessageAndLevel()`, do you think it's all right like that? Just want to minimise duplicated code
</blockquote></details>

no longer relevant comment

---

*[shared/src/main/java/com/vaadin/shared/ui/ErrorLevel.java, line 26 at r2](https://reviewable.io:443/reviews/vaadin/framework/9817#-Ks3NqQPud0MzNdGpl5Q:-KuZOx1zWmcyr1UXZ4_f:b32wtd) ([raw file](https://github.com/vaadin/framework/blob/946a741455e1da3e3609a36baa875b3245ae76fc/shared/src/main/java/com/vaadin/shared/ui/ErrorLevel.java#L26)):*
<details><summary><i>Previously, wbadam (Adam Wagner) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

ok

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Reviewed 2 of 2 files at r5.
Review status: 42 of 52 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Reviewed 3 of 34 files at r4.
Review status: 45 of 52 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

wbadam committed:
Add missing since tags

tsuoanttila commented:
Reviewed 1 of 1 files at r6.
Review status: 45 of 52 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuTMa51GZcArPMnDcPt:-KuZ_Eqfz3jjR3x9Bey-:bseb8sn) (no related file):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

Remain, marked as deprecated, moved into v7 compatibility package if possible.
</blockquote></details>

We can stop caring about this discussion for now. Will come back to this if needed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

wbadam committed:
Add error level to tab caption's tooltip

tsuoanttila commented:
Review status: 44 of 52 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kr_WChytA6b0_Ot93Fq:-KutLA-5Jp2YWEZQBCJW:bd7ncaj) (no related file):*
<details><summary><i>Previously, wbadam (Adam Wagner) wrote…</i></summary><blockquote>

Build failed but reference screenshot `LabelTooltipTest-testLabelTooltip_LINUX_PhantomJS_1_tooltipVisible.png` seems wrong to me rather than what the test produces.
</blockquote></details>

Another test failure when looking for tabsheet caption with correct error classname.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Reviewed 7 of 35 files at r1, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

wbadam committed:
Change tab sheet tooltip test to be able to find element with multiple class names

wbadam commented:
Review status: 52 of 53 files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kr_WChytA6b0_Ot93Fq:-Kuwk_GNLAWKeXEUYKI5:b-5og7na) (no related file):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

Another test failure when looking for tabsheet caption with correct error classname.
</blockquote></details>

Fixed tab sheet error tooltip test as in v7

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

Commit:
Merge remote-tracking branch 'origin/master' into wbadam-el8

Commit:
Update LabelTooltipTest screenshots

tsuoanttila commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

@hesara still going through the review once.

---

Reviewed 1 of 1 files at r8, 20 of 20 files at r9.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817#-:-KuxvuANAPPMV9oqzl1T:b-1gcgd8)*
<!-- Sent from Reviewable.io -->

hesara commented:
Did a cursory review, some comments.

---

Reviewed 4 of 35 files at r1, 2 of 16 files at r3, 8 of 34 files at r4, 1 of 1 files at r6, 16 of 20 files at r9.
Review status: all files reviewed at latest revision, 10 unresolved discussions, all commit checks successful.

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 108 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuxwh3kz2Vg434dhUaY:-Kuxwh3lOI4fY-ppTBbe:bo4l5fk) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/TooltipInfo.java#L108)):*
> ```Java
>      * @deprecated use {@link #TooltipInfo(String, ContentMode, String, Object,
>      *             ErrorLevel)} instead
>      */
> ```

could have since tag

---

*[client/src/main/java/com/vaadin/client/VCaption.java, line 785 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxwkgOjVUsDe1I8-BI:-KuxwkgOjVUsDe1I8-BJ:bccjx5x) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/VCaption.java#L785)):*
> ```Java
>         if (visible) {
>             if (errorIndicatorElement == null) {
>                 errorIndicatorElement = ErrorUtil.createErrorIndicatorElement();
> ```

Does this change the DOM for the error indicator element? Used to be a div with NBSP, now a span?

---

*[client/src/main/java/com/vaadin/client/WidgetUtil.java, line 1872 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxxcWVYgpgm4NZ67Kj:-KuxxcWVYgpgm4NZ67Kk:btrsgnj) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/WidgetUtil.java#L1872)):*
> ```Java
> 
>     /**
>      * Utility methods for displaying error message on components.
> ```

since tag

---

*[client/src/main/java/com/vaadin/client/ui/VCheckBox.java, line 116 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuxz50P3bBtTHuCbRTD:-Kuxz50P3bBtTHuCbRTE:b-s0rhc1) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/ui/VCheckBox.java#L116)):*
> ```Java
>         if (visible) {
>             if (errorIndicatorElement == null) {
>                 errorIndicatorElement = ErrorUtil.createErrorIndicatorElement();
> ```

Used to be a span with NBSP, now without?

---

*[client/src/main/java/com/vaadin/client/ui/VFormLayout.java, line 400 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxyFmVXaL_dux6Ww3R:-KuxyFmVXaL_dux6Ww3S:bnm94v7) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/ui/VFormLayout.java#L400)):*
> ```Java
>                 if (errorIndicatorElement == null) {
>                     errorIndicatorElement = ErrorUtil
>                             .createErrorIndicatorElement();
> ```

Changed in DOM from div with NBSP to a span?

---

*[client/src/main/java/com/vaadin/client/ui/VLink.java, line 159 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxyeI-F5DEYEdeSVT2:-KuxyeI0o4x-JhgyHmU_:b-kg3nus) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/ui/VLink.java#L159)):*
> ```Java
>         if (visible) {
>             if (errorIndicatorElement == null) {
>                 errorIndicatorElement = ErrorUtil.createErrorIndicatorElement();
> ```

Used to be a div, now a span?

---

*[client/src/main/java/com/vaadin/client/ui/orderedlayout/Slot.java, line 525 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxzbroW8h9wHWRSHfB:-KuxzbroW8h9wHWRSHfC:btrsgnj) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/ui/orderedlayout/Slot.java#L525)):*
> ```Java
>      *            true if the caption should be rendered as HTML, false
>      *            otherwise
>      */
> ```

since tag

---

*[server/src/main/java/com/vaadin/server/ErrorMessage.java, line 30 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuy0Pk2pUBFnFSc8Ycc:-Kuy0Pk2pUBFnFSc8Ycd:bx9ordk) ([raw file](https://github.com/vaadin/framework/blob/4d6cb7b107040ec9aedf9b73725675309a1c5729/server/src/main/java/com/vaadin/server/ErrorMessage.java#L30)):*
> ```Java
> public interface ErrorMessage extends Serializable {
> 
>     public enum ErrorLevel {
> ```

This at least should be mentioned in release notes as a breaking change, and perhaps could also break binary compatibility of add-ons that depend on a class that depends on this.

---

*[shared/src/main/java/com/vaadin/shared/AbstractComponentState.java, line 50 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuy0088SpQe-IFq7yF3:-Kuy0088SpQe-IFq7yF4:btrsgnj) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/shared/src/main/java/com/vaadin/shared/AbstractComponentState.java#L50)):*
> ```Java
>     public String errorMessage = null;
> 
>     /** Level of error */
> ```

since tag

---

*[shared/src/main/java/com/vaadin/shared/ui/tabsheet/TabState.java, line 41 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuy06d2Wg6QMx23j0v3:-Kuy06d2Wg6QMx23j0v4:bb9zmyk) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/shared/src/main/java/com/vaadin/shared/ui/tabsheet/TabState.java#L41)):*
> ```Java
>     public String key;
>     public String componentError;
>     public ErrorLevel componentErrorLevel;
> ```

javadoc and since tag

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817#-:-Kuy0ocj3FDHCUn3mZfZ:bp24bgc)*
<!-- Sent from Reviewable.io -->

wbadam committed:
Add javadoc and since tags

wbadam commented:
Review status: 49 of 54 files reviewed at latest revision, 10 unresolved discussions.

---

*[client/src/main/java/com/vaadin/client/TooltipInfo.java, line 108 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuxwh3kz2Vg434dhUaY:-Kuy2ozYX79NHj4c7jPK:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/TooltipInfo.java#L108)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

could have since tag
</blockquote></details>

Done.

---

*[client/src/main/java/com/vaadin/client/VCaption.java, line 785 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxwkgOjVUsDe1I8-BI:-Kuy2yfaDy9Y1yrkAdTQ:b-46h583) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/VCaption.java#L785)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Does this change the DOM for the error indicator element? Used to be a div with NBSP, now a span?
</blockquote></details>

Yes, it does change the DOM slightly but shouldn't mess with the look or style. Could add back `<div>` if necessary, changed for unifying reasons.

---

*[client/src/main/java/com/vaadin/client/WidgetUtil.java, line 1872 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxxcWVYgpgm4NZ67Kj:-Kuy3JCgf5YHt5i8zK-k:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/WidgetUtil.java#L1872)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

since tag
</blockquote></details>

Done.

---

*[client/src/main/java/com/vaadin/client/ui/VCheckBox.java, line 116 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuxz50P3bBtTHuCbRTD:-Kuy3NuBZFGgtt1RZrAE:b-9f3rbp) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/ui/VCheckBox.java#L116)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Used to be a span with NBSP, now without?
</blockquote></details>

Also shouldn't change the looks. I'm guessing that it used to be like that for browser compatibility.

---

*[client/src/main/java/com/vaadin/client/ui/VFormLayout.java, line 400 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxyFmVXaL_dux6Ww3R:-Kuy3_HJGTw8T09oPKId:bj9b7s6) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/ui/VFormLayout.java#L400)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Changed in DOM from div with NBSP to a span?
</blockquote></details>

Fight For Uniformity

---

*[client/src/main/java/com/vaadin/client/ui/VLink.java, line 159 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxyeI-F5DEYEdeSVT2:-Kuy4D7ZxG8MCrptYu8w:bb68p3r) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/ui/VLink.java#L159)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Used to be a div, now a span?
</blockquote></details>

#FFU

---

*[client/src/main/java/com/vaadin/client/ui/orderedlayout/Slot.java, line 525 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxzbroW8h9wHWRSHfB:-Kuy4VjdXTxpya3qyR_B:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/ui/orderedlayout/Slot.java#L525)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

since tag
</blockquote></details>

Done.

---

*[server/src/main/java/com/vaadin/server/ErrorMessage.java, line 30 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuy0Pk2pUBFnFSc8Ycc:-Kuy4ukuhfdfc7K-s_eH:b-eo87ag) ([raw file](https://github.com/vaadin/framework/blob/4d6cb7b107040ec9aedf9b73725675309a1c5729/server/src/main/java/com/vaadin/server/ErrorMessage.java#L30)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

This at least should be mentioned in release notes as a breaking change, and perhaps could also break binary compatibility of add-ons that depend on a class that depends on this.
</blockquote></details>

Should I do that? If so, where can I find the release notes?

---

*[shared/src/main/java/com/vaadin/shared/AbstractComponentState.java, line 50 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuy0088SpQe-IFq7yF3:-Kuy5CFdXCzK9l19w3jX:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/shared/src/main/java/com/vaadin/shared/AbstractComponentState.java#L50)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

since tag
</blockquote></details>

Done.

---

*[shared/src/main/java/com/vaadin/shared/ui/tabsheet/TabState.java, line 41 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuy06d2Wg6QMx23j0v3:-Kuy5b1FxHuICpyhL6pt:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/shared/src/main/java/com/vaadin/shared/ui/tabsheet/TabState.java#L41)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

javadoc and since tag
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 5 of 5 files at r10.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[client/src/main/java/com/vaadin/client/VCaption.java, line 785 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxwkgOjVUsDe1I8-BI:-KuyA1USQPJFz_sWi_oW:bh1zgip) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/VCaption.java#L785)):*
<details><summary><i>Previously, wbadam (Adam Wagner) wrote…</i></summary><blockquote>

Yes, it does change the DOM slightly but shouldn't mess with the look or style. Could add back `<div>` if necessary, changed for unifying reasons.
</blockquote></details>

I'm ok with unification if it does not change the look much, but it, too, should be mentioned in release notes as a backwards incompatibility as it affects those who have used specific theme rules.

---

*[server/src/main/java/com/vaadin/server/ErrorMessage.java, line 30 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuy0Pk2pUBFnFSc8Ycc:-KuyANQIMPcAn59naYnN:bxiw5gv) ([raw file](https://github.com/vaadin/framework/blob/4d6cb7b107040ec9aedf9b73725675309a1c5729/server/src/main/java/com/vaadin/server/ErrorMessage.java#L30)):*
<details><summary><i>Previously, wbadam (Adam Wagner) wrote…</i></summary><blockquote>

Should I do that? If so, where can I find the release notes?
</blockquote></details>

This and the DOM changes should be mentioned as incompatibilities. You can find the release notes in all/src/main/templates/release-notes.html

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

wbadam committed:
Add mention of incompatible chnages of error level and error indicator element

wbadam commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[client/src/main/java/com/vaadin/client/VCaption.java, line 785 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-KuxwkgOjVUsDe1I8-BI:-KuyFftimZQ_ttyx7IW6:bm8wc7d) ([raw file](https://github.com/vaadin/framework/blob/c866c1b1209711c8865ccc3c6ac36dc0eb877b36/client/src/main/java/com/vaadin/client/VCaption.java#L785)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

I'm ok with unification if it does not change the look much, but it, too, should be mentioned in release notes as a backwards incompatibility as it affects those who have used specific theme rules.
</blockquote></details>

Mentioned in release notes

---

*[server/src/main/java/com/vaadin/server/ErrorMessage.java, line 30 at r9](https://reviewable.io:443/reviews/vaadin/framework/9817#-Kuy0Pk2pUBFnFSc8Ycc:-KuyFjlg55aTbwJNmyX4:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/4d6cb7b107040ec9aedf9b73725675309a1c5729/server/src/main/java/com/vaadin/server/ErrorMessage.java#L30)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

This and the DOM changes should be mentioned as incompatibilities. You can find the release notes in all/src/main/templates/release-notes.html
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817)*
<!-- Sent from Reviewable.io -->

wbadam committed:
Fix typo

hesara commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r11.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9817#-:-KuyHOdAwjuOUI9XfMQn:bnfp4nl)*
<!-- Sent from Reviewable.io -->
